src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -203,9 +203,11 @@ private void TranscodeAndWriteRawValue(ReadOnlySpan<char> json, bool skipInputVa
             byte[]? tempArray = null;
 
             // For performance, avoid obtaining actual byte count unless memory usage is higher than the threshold.
-            Span<byte> utf8Json = json.Length <= (JsonConstants.ArrayPoolMaxSizeBeforeUsingNormalAlloc / JsonConstants.MaxExpansionFactorWhileTranscoding) ?
-                // Use a pooled alloc.
-                tempArray = ArrayPool<byte>.Shared.Rent(json.Length * JsonConstants.MaxExpansionFactorWhileTranscoding) :
+            Span<byte> utf8Json =
+                // Use stack memory
+                json.Length <= (JsonConstants.StackallocByteThreshold / JsonConstants.MaxExpansionFactorWhileTranscoding) ? stackalloc byte[JsonConstants.StackallocByteThreshold] :
+                // Use a pooled array
+                json.Length <= (JsonConstants.ArrayPoolMaxSizeBeforeUsingNormalAlloc / JsonConstants.MaxExpansionFactorWhileTranscoding) ? tempArray = ArrayPool<byte>.Shared.Rent(json.Length * JsonConstants.MaxExpansionFactorWhileTranscoding) :
                 // Use a normal alloc since the pool would create a normal alloc anyway based on the threshold (per current implementation)
                 // and by using a normal alloc we can avoid the Clear().
                 new byte[JsonReaderHelper.GetUtf8ByteCount(json)];
```

Comment:
Just curious, are we interested in making some sort of helper for this pattern? I remember we discussed somewhere an extra C# support for it, but, Ipresume, it can also be done with runtime-only changes, to look like

```cs
(scoped?) PooledSpan<byte> ps = PooledSpan.Rent(len);

use(ps.Span);

ps.Return();
```
and expand that `Rent` in runtime as localloc/ArrayPool/newarr depending what's more profitable (and e.g. avoid using localloc if Rent is inside a loop, etc).

Reply:
We first need the ability for that "localloc" inside of that hypothetical Rent to be able to allocate in the context of its caller.